packages/frontend-shared/src/gql-components/HeaderBarContent.vue

```diff
@@ -152,9 +168,14 @@ fragment HeaderBar_HeaderBarContent on Query {
   projectRootFromCI
   ...TopNav
   ...Auth
+  ...HeaderBarContent_Auth
 }
 `
 
+const userData = computed(() => {
+  return props.gql.cloudViewer ?? props.gql.cachedUser
+})
```

Comment:
what do you think about doing this on the server-side instead? IMO the HeaderBarContent component shouldn't have to understand the inner workings of how we cache user data

Reply:
I would definitely opt for that if there was a good way. With how I've written it, it currently can't happen b/c the `cloudViewer` is resolved directly from the cloud, so it returns `null` immediately when the type isn't fulfilled with the full shape (and there is more info that doesn't exist in the `CachedUser`). The `cachedUser` type we expose in the GraphQL schema is to avoid the flash that happens when navigating pages.

The component is only aware so far as what info we provide via the graphql schema, so it's technically user-facing in that sense, but I agree it's still not great.

I'll think a bit about whether there's a better solution here, e.g. by having a way to provide an alternate result on cache miss for `deferIfNotLoaded` though 🤔

Reply:
Yeah, I think this is probably the simplest solution for now honestly. Anything else would add a lot more complexity for what is effectively an edge case... again, it's exposed in the GraphQL schema, so that contract is as far as the client knows about anything there